ssl/ssl_lib.c

```diff
@@ -7211,3 +7260,220 @@ int SSL_stream_conclude(SSL *ssl, uint64_t flags)
     return 0;
 #endif
 }
+
+int SSL_add_expected_rpk(SSL *s, EVP_PKEY *rpk, int nid)
+{
+    unsigned char *i2dbuf = NULL;
+    SSL_DANE *dane = SSL_get0_dane(s);
+    size_t i2dlen;
+    unsigned char dgstbuf[EVP_MAX_MD_SIZE];
+    unsigned int dgstlen;
+    unsigned char *data;
+    size_t len;
+    int ret;
+    const EVP_MD *md = NULL;
+    int matching;
+    int ok = 0;
+
+    if (dane == NULL)
+        return 0;
+
+    /*
+     * In practice, the OSSL_RPK_MATCH constants
+     * match the DANETLS_MATCHING constants, but
+     * there's always a chance they may diverge
+     * as new values are added. So, do a translation
+     */
+    switch (nid) {
+    case OSSL_RPK_MATCH_FULL:
+        matching = DANETLS_MATCHING_FULL;
```

Comment:
If the `DANETLS_MATCH_*` constants are in a non-public header, we should fix that, and avoid the need to apply a conversion (that is actually the identity function)?  Then any (unsigned 8-bit) value that is either 0 or has a slot in the DANE mtype table, works, with no explicit set of known values here.

Reply:
They are in a private header, and I didn't want to touch them. But it is a valid option.
Yes, it's an identity function, but could be non-identity if things changed in the future (not sure they would).